src/hooks/util/CommentLabelExtractor.ts

```diff
@@ -0,0 +1,57 @@
+import { ESTree } from "meriyah";
+import { SourceMapConsumer } from "source-map-js";
+
+export default class CommentLabelExtractor {
+  private lineToComment = new Map<number, ESTree.Comment | undefined>();
+
+  constructor(
+    public comments: ESTree.Comment[],
+    public sourceMap?: SourceMapConsumer,
+  ) {}
+
+  private locate(loc: ESTree.Node["loc"]): number | undefined {
+    if (!loc) return undefined;
+    if (this.sourceMap) {
+      const mapped = this.sourceMap.originalPositionFor(loc.start);
+      if (mapped?.line) return mapped.line;
+    } else return loc.start.line;
+  }
+
+  private findCommentInLine(line: number) {
+    if (!this.lineToComment.has(line)) {
+      const comment = this.comments.find(
+        (c) => this.locate(c.loc) == line && c.type == "SingleLine",
+      );
+      this.lineToComment.set(line, comment);
+    }
+
+    return this.lineToComment.get(line);
+  }
+
+  labelsFor(funStartLine: number) {
+    let labels: string[] | undefined;
+
+    // @label or @labels can be specified in a single line comment
+    // within continuous single line comments preceding a function:
```

Comment:
I thought this edge case would complicate the code a bit:

```javascript
/* @label l1 */ function f() { ... }
function g() { ... }
```

With the current simple approach label `l1` will be applied to function `g` instead of `f`. 

Assuming `acorn-walk` reports the nodes in lexical order, we can assign comments to functions in FIFO order and cover this edge case. But we need to do this for every function even if it's excluded from the instrumentation. I can do this of course. What do you think?

Reply:
Eh, probably not worth the trouble.